homeassistant/components/mystrom/switch.py

```diff
@@ -22,30 +24,34 @@
 )
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Find and return myStrom switch."""
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the myStrom switch/plug integration."""
     name = config.get(CONF_NAME)
     host = config.get(CONF_HOST)
 
     try:
-        MyStromPlug(host).get_status()
+        plug = _MyStromSwitch(host)
+        await plug.get_state()
     except MyStromConnectionError:
-        _LOGGER.error("No route to device: %s", host)
+        _LOGGER.error("No route to myStrom plug: %s", host)
         raise PlatformNotReady()
 
-    add_entities([MyStromSwitch(name, host)])
+    async_add_entities([MyStromSwitch(plug, name, plug.mac)])
```

Comment:
Why do we pass `plug.mac` to the class, if we also pass `plug` in?

Reply:
Because I thought that it would be more clear what's happening. Switched to use `plug`.